circadian/models.py

```diff
@@ -1,878 +1,950 @@
+    LightPhase = self.sigma * Bhat - A1_term_phase - A2_term_phase
+
+    dydt = np.zeros_like(state)
+    dydt[0,...] = -1.0 * self.gamma * R + self.K * np.cos(self.Beta1) / 2.0 * R * (1.0 - pow(R, 4.0)) + LightAmp
+    dydt[1,...] = 2*np.pi/self.tau + self.K / 2.0 * np.sin(self.Beta1) * (1 + pow(R, 4.0)) + LightPhase
+    dydt[2,...] = 60.0 * (alpha * (1.0 - n) - self.delta * n)
+
+    return dydt
+
+# %% ../nbs/00_models.ipynb 62
+@patch_to(Hannay19)
+def phase(self,
+          trajectory: DynamicalTrajectory=None, # trajectory to calculate the phase. If None, the current trajectory is used
+          time: float=None # a time point to calculate the phase at. If None, the phase is calculated for the entire trajectory
+          ) -> float:
+    if trajectory is None:
+        trajectory = self.trajectory
+    else: 
+        if not isinstance(trajectory, DynamicalTrajectory):
+            raise ValueError("trajectory must be a DynamicalTrajectory")
+    if time is None:
+        x = np.cos(trajectory.states[:, 1])
+        y = np.sin(trajectory.states[:, 1])
+    else:
+        if not isinstance(time, (float, int)):
+            raise ValueError("time must be a float or an int")
+        else:
+            state = trajectory(time)
+            x = np.cos(state[1])
+            y = np.sin(state[1])
+    return np.angle(x + complex(0,1) * y)
+
+# %% ../nbs/00_models.ipynb 64
+@patch_to(Hannay19)
+def amplitude(self,
+              trajectory: DynamicalTrajectory=None, # trajectory to calculate the amplitude. If None, the current trajectory is used
+              time: float=None # a time point to calculate the amplitude at. If None, the amplitude is calculated for the entire trajectory
+              ) -> float:
+    if trajectory is None:
+        trajectory = self.trajectory
+    else: 
+        if not isinstance(trajectory, DynamicalTrajectory):
+            raise ValueError("trajectory must be a DynamicalTrajectory")
+    if time is None:
+        amplitude = trajectory.states[:, 0]
+    else:
+        if not isinstance(time, (float, int)):
+            raise ValueError("time must be a float or an int")
+        else:
+            state = trajectory(time)
+            amplitude = state[0] 
+    return amplitude
+
+# %% ../nbs/00_models.ipynb 66
+@patch_to(Hannay19)
+def cbt(self,
+        trajectory: DynamicalTrajectory=None # trajectory to calculate the cbt. If None, the current trajectory is used
+        ) -> np.ndarray:
+    "Finds the core body temperature minumum markers for the model along a trajectory as the times where the phase is pi"
+    if trajectory is None:
+        trajectory = self.trajectory
+    else:
+        if not isinstance(trajectory, DynamicalTrajectory):
+            raise ValueError("trajectory must be a DynamicalTrajectory")
+    inverted_x = -np.cos(trajectory.states[:,1])
+    cbt_min_idxs, _ = find_peaks(inverted_x)
```

Comment:
Investigate find_peaks' behavior when there are multiple peaks in rapid succession: Does it deduplicate in some way?

Reply:
Added a `_check_cbtmin_spacing()` function that throws a warning if any difference in cbtmin is less than 6 hours